test/helpers/kubectl.go

```diff
@@ -1013,52 +1032,63 @@ func addIfNotOverwritten(options []string, field, value string) []string {
 	return options
 }
 
-// ciliumInstallHelm installs Cilium with the Helm options provided.  Returns an
-// error if any patch or if any original descriptors files were not found.
-func (kub *Kubectl) ciliumInstallHelm(options []string) error {
-	defaultOptions := map[string]string{
-		"global.registry":               "k8s1:5000/cilium",
-		"agent.image":                   "cilium-dev",
-		"agent.tag":                     "latest",
-		"operator.image":                "operator",
-		"operator.tag":                  "latest",
-		"managed-etcd.registry":         "docker.io/cilium",
-		"global.debug":                  "true",
-		"global.k8s.requireIPv4PodCIDR": "true",
-		"global.pprof.enabled":          "true",
-		"global.logSystemLoad":          "true",
-		"global.bpf.preallocateMaps":    "true",
-		"global.etcd.leaseTTL":          "30s",
-		"global.ipv4.enabled":           "true",
-		"global.ipv6.enabled":           "true",
-	}
-
-	for key, value := range defaultOptions {
+func (kub *Kubectl) generateCiliumYaml(options []string, filename string) error {
+	for key, value := range defaultHelmOptions {
 		options = addIfNotOverwritten(options, key, value)
 	}
 
 	// TODO GH-8753: Use helm rendering library instead of shelling out to
 	// helm template
-	res := kub.ExecMiddle(fmt.Sprintf("helm template %s --namespace=kube-system %s > cilium.yaml",
-		HelmTemplate, strings.Join(options, " ")))
+	res := kub.ExecMiddle(fmt.Sprintf("helm template %s --namespace=kube-system %s > %s",
+		HelmTemplate, strings.Join(options, " "), filename))
 	if !res.WasSuccessful() {
 		return res.GetErr("Unable to generate YAML")
 	}
 
-	res = kub.Apply("cilium.yaml")
+	return nil
+}
+
+// ciliumInstallHelm installs Cilium with the Helm options provided.  Returns an
+// error if any patch or if any original descriptors files were not found.
+func (kub *Kubectl) ciliumInstallHelm(options []string) error {
+	if err := kub.generateCiliumYaml(options, "cilium.yaml"); err != nil {
+		return err
+	}
+
+	res := kub.Apply("cilium.yaml")
 	if !res.WasSuccessful() {
 		return res.GetErr("Unable to apply YAML")
 	}
 
 	return nil
 }
 
+// ciliumUninstallHelm installs Cilium with the Helm options provided.  Returns an
+// error if any patch or if any original descriptors files were not found.
+func (kub *Kubectl) ciliumUninstallHelm(options []string) error {
+	if err := kub.generateCiliumYaml(options, "cilium.yaml"); err != nil {
+		return err
+	}
+
+	res := kub.Delete("cilium.yaml")
```

Comment:
Do we generate different resources in K8s based off of the Helm options? Most of these appear to be options provided to the DaemonSet / ConfigMap - can you elaborate on what K8s resources are different depending upon the Helm options?

Reply:
* Inclusion of etcd-operator and related ClusterRole, ClusterRoleBinding and ServiceAccount.